newton/sim/ik.py

```diff
@@ -920,57 +957,40 @@ def _update_position_targets(
 def _pos_jac_analytic(
     link_index: int,
     link_offset: wp.vec3,
-    joint_child: wp.array1d(dtype=wp.int32),  # (n_joints)
-    coord_to_joint: wp.array1d(dtype=wp.int32),  # (n_coords)
-    affects_coord: wp.array1d(dtype=wp.uint8),  # (n_coords)
-    joint_qd_start: wp.array1d(dtype=wp.int32),  # (n_joints + 1)
-    joint_q_start: wp.array1d(dtype=wp.int32),  # (n_joints + 1)
-    joint_type: wp.array1d(dtype=wp.int32),  # (n_joints)
-    joint_S_s: wp.array2d(dtype=wp.spatial_vector),  # (n_problems, n_dofs)
+    affects_dof: wp.array1d(dtype=wp.uint8),  # (n_dofs)
     body_q: wp.array2d(dtype=wp.transform),  # (n_problems, n_bodies)
+    joint_S_s: wp.array2d(dtype=wp.spatial_vector),  # (n_problems, n_dofs)
     start_idx: int,
-    n_coords: int,
+    n_dofs: int,
     weight: float,
-    # outputs
-    jacobian: wp.array3d(dtype=wp.float32),  # (n_problems, n_residuals, n_coords)
+    # output
+    jacobian: wp.array3d(dtype=wp.float32),  # (n_problems, n_residuals, n_dofs)
 ):
-    global_coord = wp.tid()
-
-    problem_idx = global_coord // n_coords
-    coord_idx = global_coord % n_coords
+    # one thread per (problem, dof)
+    global_dof = wp.tid()
```

Comment:
Can you use `(problem, dof) = wp.tid()` directly and launch the kernel with a 2-dimensional `dim`?